src/NServiceBus.Core/Routing/Routers/UnicastSendRouterConnector.cs

```diff
@@ -41,7 +45,10 @@ public enum RouteOption
             {
                 throw new InvalidOperationException("Cannot route to a specific instance because an endpoint instance discriminator was not configured for the destination endpoint. It can be specified via EndpointConfiguration.MakeInstanceUniquelyAddressable(string discriminator).");
             }
-            var thisEndpoint = state.Option == RouteOption.RouteToAnyInstanceOfThisEndpoint ? sharedQueue : null;
+
+            var effectiveSharedQueue = ApplyDistributorLogic(context);
+
+            var thisEndpoint = state.Option == RouteOption.RouteToAnyInstanceOfThisEndpoint ? effectiveSharedQueue : null;
```

Comment:
why do we need to apply the distributor address here in the first place? This will only be applied when using the explicit RouteToAnyInstanceOfThisEndpoint send option. Doesn't seem to be directly related to our bug?